# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannel.h
# Commit: 64e118a38c1d
# Full Hash: 64e118a38c1d4ae364b6ddfc618909546c1b92f2
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-07-21 09:42:41
# Regressor Bug: 1646899
# File Overlap Count: 1
# Description:
#   Bug 1646899 - P3: Modify CanUseDocumentChannel to take necessary vars. r=mattwoodrow
#   
#   Allows checking if DocumentChannel can be used with out creating an
#   nsDocShellLoadState object.
#   
# ==============================================================================

diff -r fc96ea4de326 -r 64e118a38c1d netwerk/ipc/DocumentChannel.h
--- a/netwerk/ipc/DocumentChannel.h	Tue Jul 21 01:01:00 2020 +0000
+++ b/netwerk/ipc/DocumentChannel.h	Tue Jul 21 01:01:03 2020 +0000
@@ -64,7 +64,7 @@
       nsLoadFlags aLoadFlags, nsIInterfaceRequestor* aNotificationCallbacks,
       uint32_t aCacheKey, bool aUriModified, bool aIsXFOError);
 
-  static bool CanUseDocumentChannel(nsDocShellLoadState* aLoadState);
+  static bool CanUseDocumentChannel(nsIURI* aURI, uint32_t aLoadFlags);
 
  protected:
   DocumentChannel(nsDocShellLoadState* aLoadState, class LoadInfo* aLoadInfo,
